classes/stock/StockAvailable.php

```diff
@@ -698,6 +740,33 @@ public static function outOfStock($id_product, $id_shop = null)
         return (int) Db::getInstance(_PS_USE_SQL_SLAVE_)->getValue($query);
     }
 
+    /**
+     * @param int $id_product
+     * @param int id_product_attribute Optional
+     * @param int $id_shop Optional
+     *
+     * @return bool|string
+     */
+    public static function getLocation($id_product, $id_product_attribute = null, $id_shop = null)
```

Comment:
$id_shop should be default evaluated to 0, same for $id_product_attribute.

Reply:
Would it introduce some magic numbers inside this function ? I know in PrestaShop ID 0 is the default one. However adding it as default value would hard-code this rule into the code. What if one day we want to move to uuid SQL IDs ? Even if we extend StockAvailable, the value `0` will be hardcoded into the function signature.

So ... what about `null` as default value always :) ?

Reply:
if we move from ids to uuid we will have bigger problems as we expected integer or null here and we will change the functions to use some Uuid objects, isn't it?

It's only 1 implementation, I don't think we will break it in 1.7.x release

Reply:
Even if we use only one type, we are able to set a value to `null`. Which is logic to have when there is no id attached to the object, because 0 is an id :)
If we move to uuid or anything else, it's easier to test if value is null instead of checking if it's numeric and not 0 (and it costs more resources).

```php
<?php

class A
{
    public function __construct(int $a)
    {
        var_dump($a);
    }
}

new A();
new A(22);
new A('azerty');
```

Will produce:
```bash
$ php7.2 test.php 
/home/prambaud/test.php:7:
NULL
/home/prambaud/test.php:7:
int(22)
PHP Fatal error:  Uncaught TypeError: Argument 1 passed to A::__construct() must be of the type integer or null, string given, called in /home/prambaud/test.php on line 13 and defined in /home/prambaud/test.php:5
Stack trace:
#0 /home/prambaud/test.php(13): A->__construct('azerty')
#1 {main}
  thrown in /home/prambaud/test.php on line 5
```